R/model-module-helpers.R

```diff
@@ -178,23 +178,16 @@ add_pmfs <- function(pmfs) {
   if (!is.list(pmfs)) {
     return(pmfs)
   }
-  lpmfs <- purrr::map_dbl(pmfs, length)
-  l <- sum(lpmfs)
-  conv <- rep(0, l)
-  conv[1:lpmfs[1]] <- pmfs[[1]]
-  for (s in 2:d) {
-    # P(Z = z) = sum_over_x(P(X = x) * P(Y = z - x)) # nolint
-    proc <- rep(0, l)
-    for (i in 1:l) {
-      for (j in 1:lpmfs[s]) {
-        if (i >= j) {
-          proc[i] <- proc[i] + pmfs[[s]][j] * conv[i - j + 1]
-        }
-      }
+  conv <- Reduce(x = pmfs, f = function(conv, pmf) {
```

Comment:
Is the assignment / last line thing indicated by the styler? Seems weird, but I'm terse.

I prefer the bare "last evaluation" return approach (though not as dev friendly); if not that, wrap in `return(...)` (more dev friendly, marginal performance hit). Assignment is an unnecessary symbol allocation / parsing step.

Reply:
It isn't - that's just my concession to readability - happy to wrap the `Reduce` in a `return` if that's more the package style

Reply:
I'd prefer to see returns used to match package style.

Reply:
Fixed in https://github.com/epinowcast/epinowcast/pull/187/commits/cc5ecb4e92671dc81d46544f923c38cc24ff6b4a

Reply:
The internal `return` is still missing and to be honest, I would prefer the slight performance hint of having an assignment followed by a return. I'm happy to read links that argue against that as a stylistic choice of course.